### DISCUSSION TITLE
Wrong Pins assigned to ESP C3 32s

### DISCUSSION THREAD

**Discussion by Erkman84** [micropython/micropython#15831] [ESP32]
Hi there,
It's about this:
![ai_thinker_esp32-c3s](https://github.com/user-attachments/assets/df4d3ead-8c64-4dd4-88e9-f5f07da9bb74)

For downloading anf flashing the firmware there was only one possibility: ESP32_GENERIC_C3

I've tried to use the pins, especially the ADCs, and then I run in this bug:
I know that the Pin 18-19 are not working previously (maybe for USB or something) read somewhere, but that's okay.
But, what was not: The Pin 0 is connected to ADC Pin and GPIO 0. 
Next: Pin 2, 3, 4, 5 is working as expected (turning the onboard bgr leds on/off) that is also okay.
What I was wondering: Pin(10) --> is Pin 12
Pin 9 is nowhere
Pin(8) --> is Pin 10
Pin 6,7 is nowhere

Pins above 10 cannot be found at any free digital ports. Where are they?

So most of the Pins (output/input) you cannot use with this firmware. Is it a bug or feature?
Tested with common LED blinking.

**Comment by robert-hh** [comment]
The Pin number of the ESP32 ports are just the GPIO numbers. There is not MCU pin to board pin mapping like in other ports. If they do not match with your board, the board wiring is wrong. If you cite Pin(10) is Pin 12, what is printed on the board's silk screen of this Pin?

**Reply by Erkman84** [reply]
Yes, sorry, recap the back of the mcu and relocated the "real pins" so found out, that IO18, IO19, IO9 are not working only. This does not look so bad... other pins are special pins ex SPI and UART, thats a little pity.
For others: the IO Pins are useable: 1, 2, 3, 4, 5, 6, 7, 8 and 10 for digital out.
![20240911_114920](https://github.com/user-attachments/assets/47d8d1a7-3807-4b6c-8213-50dffea4e961)

**Comment by Erkman84** [comment]
Another problem, to determine the "other" Pins:
help(Pin.board) does not work...
`help(Pin.board)
object <class 'board'> is of type type`
and
`dir(Pin.board)
['__class__', '__name__', '__bases__', '__dict__']`

**Reply by Josverl** [reply]
Due to size constraints, Micropython does not include Docstrings, and thus the help function can't display them 

In stead you can use the documentation, which I have also integrated with type stubs, that most IDEs can also use for 'while-coding' lookups 

I _think_ I've published stubs for the generic ESP32_C3 board, but unfortunately they still won't have the board/silkscreen labels that are printed on your particular board

**Reply by robert-hh** [reply]
That's intentional for the GENERIC boards, which assume that these label the GPIO numbers at the board. Then a pin mapping file pins.csv does not exist and hence the Pin.board class is empty. For your specific board you have to test yourself by trying, which GPIO numbers match to the SPIxx and U0x pins. For the U0RX it should be GPIO20, for U0TX GPIO21, for SPIxx GPIO14 - GPIO17.  SPICS0 -> GPIO14, PICLK -> GPIO15, SPID -> GPIO16, SPIQ -> GPIO17,
That matches the Pin names of the data sheet.

**Reply by Erkman84** [reply]
Thank you guys, I will try them and give you a feedback.
